sklearn/preprocessing/_function_transformer.py

```diff
@@ -59,23 +62,60 @@ class FunctionTransformer(BaseEstimator, TransformerMixin):
 
         .. deprecated::0.19
 
+    check_inverse : bool, (default=False)
+       Whether to check that ``transform`` followed by ``inverse_transform``
+       or ``func`` followed by ``inverse_func`` leads to the original targets.
+
+       .. versionadded:: 0.20
+
     kw_args : dict, optional
         Dictionary of additional keyword arguments to pass to func.
 
     inv_kw_args : dict, optional
         Dictionary of additional keyword arguments to pass to inverse_func.
 
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by np.random. Note that this is used to compute if func and
+        inverse_func are the inverse of each other.
+
+
     """
     def __init__(self, func=None, inverse_func=None, validate=True,
-                 accept_sparse=False, pass_y='deprecated',
-                 kw_args=None, inv_kw_args=None):
+                 accept_sparse=False, pass_y='deprecated', check_inverse=False,
+                 kw_args=None, inv_kw_args=None, random_state=None):
         self.func = func
         self.inverse_func = inverse_func
         self.validate = validate
         self.accept_sparse = accept_sparse
         self.pass_y = pass_y
+        self.check_inverse = check_inverse
         self.kw_args = kw_args
         self.inv_kw_args = inv_kw_args
+        self.random_state = random_state
+
+    def _validate_inverse(self, X):
+        """Check that func and inverse_func are the inverse."""
+        # Apply the transform and inverse_transform on few samples.
+        X = check_array(X, accept_sparse='csr')
+        random_state = check_random_state(self.random_state)
+        n_subsample = min(100, X.shape[0])
+        subsample_idx = random_state.choice(range(X.shape[0]),
+                                            size=n_subsample,
+                                            replace=False)
+        if sparse.issparse(X):
+            X_sel = X[subsample_idx].A
```

Comment:
I don't think you can be sure that the function accepts dense data, so you can't do this.